Source/SimpleActions/Public/SimpleAction.h

```diff
@@ -60,17 +84,53 @@ class SIMPLEACTIONS_API USimpleAction : public UObject, public FTickableGameObje
 	TWeakObjectPtr<AActor> ActingActor = nullptr;
 	
 	// The object which triggered this action.
-	UPROPERTY(BlueprintReadOnly, Transient, Category = "Simple Action")
+	UPROPERTY(BlueprintReadOnly, Transient,Category = "Simple Action")
 	TWeakObjectPtr<UObject> ActionInstigator = nullptr;
 
+	// The amount of time before the Action Starts.
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Simple Action")
+	float StartDelay;
+
+	// The amount of time the action will be forced to last.
+	UPROPERTY(EditAnywhere, BlueprintReadWrite, Category = "Simple Action")
+	float DurationOverride;
+
 	// Called when the action is started.
 	UPROPERTY(BlueprintAssignable, Category = "Simple Action")
 	FOnActionStartedSignature OnActionStarted;
+
+	// Called when the action in Interrupted.
+	UPROPERTY(BlueprintAssignable, Category = "Simple Action")
+	FOnActionInterruptedSignature OnActionInterrupted;
```

Comment:
Should keep terminology consistent: `OnActionCanceled`.

Reply:
There should be both eventually but it's not in the scope of the current task.
Cancel = Deliberate stopping of an action.
Interrupt = Another action has started that does not allow this action to continue.
I'll rename this one as Cancel as that's the current functionality and throw in a note for future interrupt state.